src/nvim/tui/tui.c

```diff
@@ -438,15 +439,15 @@ static void sigwinch_cb(SignalWatcher *watcher, int signum, void *data)
 static bool attrs_differ(HlAttrs a1, HlAttrs a2, bool rgb)
 {
   if (rgb) {
-    // TODO(bfredl): when we start to support special color,
-    // rgb_sp_color must be added here
     return a1.rgb_fg_color != a2.rgb_fg_color
       || a1.rgb_bg_color != a2.rgb_bg_color
-      || a1.rgb_ae_attr != a2.rgb_ae_attr;
+      || a1.rgb_ae_attr != a2.rgb_ae_attr
+      || a1.rgb_sp_color != a2.rgb_sp_color;
   } else {
     return a1.cterm_fg_color != a2.cterm_fg_color
       || a1.cterm_bg_color != a2.cterm_bg_color
-      || a1.cterm_ae_attr != a2.cterm_ae_attr;
+      || a1.cterm_ae_attr != a2.cterm_ae_attr
+      || a1.rgb_sp_color != a2.rgb_sp_color;
```

Comment:
Do we need `cterm_sp_color` ? cc @teto @bfredl

Reply:
I would expect colored underline is only supported in terminals that support rgb anyway, so we could make this feature exclusive to rgb mode (`'termguicolors'`).

Reply:
But then you would need to use`ctermfg/bg` to configure text, and `guisp` to configure underlines. Seems a little weird (in the past `cterm*` and `gui*` have been mutually exclusive configurations), though there is no strict reason we need to forbid it.

Reply:
Yeah.

I am not sure if adding `ctermsp` makes sense "just" for that though.
And if so, it would be nice to default to `guisp` then anyway, since this is defined in colorschemes already.

Reply:
Sure, what we could do then is to check if `a1.cterm_ae_attr` matches `HL_UNDERLINE|HL_UNDERCURL`, because otherwise `rgb_sp_color` has no effect in cterm mode.